### ISSUE TITLE
Introduce project settings for enabling prebuilds for branches, pull requests, etc.

### ISSUE THREAD

**Issue by gtsiolis** [gitpod-io/gitpod#7426]
### Problem to solve

Following the problem statement in https://github.com/gitpod-io/gitpod/issues/7425, :a: prebuilds are only enable for Github repositories for the default branch and :b:  prebuilds are not enabled when adding a repository as project.

> For GitHub repositories, when installing the GitHub App, prebuilds are enabled by default for the default branch. For enabling prebuilds for all branches, a user will need to defien this in `.gitpod.yml`, see relevant docs.

> For GitLab and Bitbucket repositories, when adding aproject a webhook is installed. However, enabled are not enabled, see #7424 and pending PR in #7422. Prebuilds are also not enabled when adding a project, following the pattern used for GitHub repositories.

> This makes enabling prebuilds for all branches a less discoverable option and more difficult to set up.

### Proposal

Introduce a preference in project settings that allows enabling or disabling prebuilds for a) branches, b) pull requests, c)  pull requests from forks, add check, add comment, and add badge.

**Comment by jldec** [comment]
thank you @gtsiolis - i have added this to our prebuilds investigations / backlog.

**Comment by jldec** [comment]
@gtsiolis would you mind correcting the description above given that new projects do enable prebuilds across git providers now.

Also regardig the proposal, could you please explain the benefit of moving this configuration out of .gitpod.yml and into the project settings? I would suggest that simply removing the "github" scope of the current configuration, but leaving it in .gitpod.yml just like today is a simpler, and more portable solution.

**Comment by jmls** [comment]
I would like to have an option to configure prebuilds per branch. Running into some really strange errors starting up a workspace if there is a prebuild running (can't find a user variable defined as available to scope */*

**Comment by jldec** [comment]
jmls - if you can repro the strange errors you mentioned ☝️ or still have it open, would you mind providing some more details? (repo / project / name of workspace / copy of error message) thanks!

**Comment by apolopena** [comment]
Yeah I work on branches testing gitpod configurations quite often and having prebuilds turned off for gitpod workspace images increases testing time by 100%. This feature would be really nice to have.

 In fact correct me if I am wrong  but I remember a time when all gitpod workspace images were prebuilt by default no matter what.

**Comment by shaal** [comment]
Perhaps add the label `meta: never-stale` ?
